.github/workflows/build_and_run_test_app.yml

```diff
@@ -0,0 +1,111 @@
+name: Build and Run Test Application
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    name: Build Test App
+    strategy:
+      matrix:
+        idf_ver: ["release-v4.3", "release-v4.4", "latest"]
+        idf_target: ["esp32", "esp32c3", "esp32s3"] # @todo ESP32S2 has less RAM and the test_app will not fit
+        exclude:
+          - idf_ver: "release-v4.3"
+            idf_target: esp32s3 # ESP32S3 support started with version 4.4
+    runs-on: ubuntu-20.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: 'true'
+      - name: Build Test Application
+        env:
+          IDF_TARGET: ${{ matrix.idf_target }}
+        shell: bash
+        working-directory: test_app
+        run: |
+          . ${IDF_PATH}/export.sh
+          idf.py build
+      - name: Merge binaries
+        working-directory: test_app/build
+        run: |
+          pip install esptool
+          python -m esptool --chip esp32 merge_bin --fill-flash-size 4MB -o flash_image.bin @flash_args
+      - uses: actions/upload-artifact@v2
+        with:
+          name: test_app_bin_${{ matrix.idf_target }}_${{ matrix.idf_ver }}
+          path: test_app/build/flash_image.bin
```

Comment:
For my comment above of `pytest-embedded` is able to flash the binary, we gonna have to upload the `flasher_args.json` file too.